autoload/wstrip.vim

```diff
@@ -22,15 +22,25 @@ function! s:is_git_repo() abort
 endfunction
 
 
+function! s:is_tracked() abort
+  call system(printf('git ls-files --error-unmatch "%s"', expand('%')))
+  return !v:shell_error
+endfunction
+
+
 function! s:get_diff_lines() abort
-  if s:is_git_repo() && executable('git')
+  if s:is_git_repo() && executable('git') && s:is_tracked()
     let cmd = 'git diff -U0 --no-ext-diff HEAD:"%s" "%s"'
   elseif executable('diff')
     let cmd = 'diff -U0 "%s" "%s"'
   else
     return []
   endif
 
+  if !isdirectory(expand('%:p:h'))
```

Comment:
This deals with files that are being edited in non-existing directories. Saving such files results in error:

```
"a/b/c" E212: Can't open file for writing
Press ENTER or type command to continue
```

However, without this check, it would fail with:
```
Error detected while processing function wstrip#auto[2]..wstrip#clean[12]..<SNR>96_get_diff_lines:
line    3:
E114: Missing quote: "!isdirectory(expand('%:p:h'))
E15: Invalid expression: empty(bufname('%')) || "!isdirectory(expand('%:p:h'))
Error detected while processing function wstrip#auto[2]..wstrip#clean:
line   12:
"a/b/c"
"a/b/c" E212: Can't open file for writing
Press ENTER or type command to continue
```

That wieldy error looks like it's somehow wstrip.vim's fault, but it isn't.